src/garage/np/policies/policy.py

```diff
@@ -20,77 +12,84 @@ def get_action(self, observation):
         Args:
             observation (np.ndarray): Observation from the environment.
 
-        Returns:
-            (np.ndarray): Action sampled from the policy.
+        """
+
+    @abc.abstractmethod
+    def get_actions(self, observations):
+        """Get actions given observations.
+
+        Args:
+            observations (torch.Tensor): Observations from the environment.
 
         """
 
-    def reset(self, dones=None):
+    def reset(self, do_resets=None):
         """Reset the policy.
 
-        If dones is None, it will be by default np.array([True]) which implies
-        the policy will not be "vectorized", i.e. number of parallel
-        environments for training data sampling = 1.
+        This is effective only to recurrent policies.
+
+        do_resets is an array of boolean indicating
+        which internal states to be reset. The length of do_resets should be
+        equal to the length of inputs, i.e. batch size.
 
         Args:
-            dones (numpy.ndarray): Bool that indicates terminal state(s).
+            do_resets (numpy.ndarray): Bool array indicating which states
+                to be reset.
 
         """
 
     @property
     def observation_space(self):
         """akro.Space: The observation space of the environment."""
-        return self._env_spec.observation_space
 
     @property
     def action_space(self):
         """akro.Space: The action space for the environment."""
-        return self._env_spec.action_space
 
-    def log_diagnostics(self, paths):
-        """Log extra information per iteration based on the collected paths.
+    @property
+    def env_spec(self):
```

Comment:
Honestly not convinced this property should exist here. It's only used in the TensorFlow branch, so why not put it on `garage.tf.Policy` instead?

Reply:
> It's good to see this interface finally coming together!

Thanks for your comments! I'm not familiar with torch tree so your comments are really important and helpful.